### ISSUE TITLE
Reflect SPV_REFLECT_DESCRIPTOR_TYPE_SAMPLE_WEIGHT_IMAGE_QCOM when decoration_flag contains SPV_REFLECT_DECORATION_WEIGHT_TEXTURE

### ISSUE THREAD

**Issue by raegnar** [KhronosGroup/SPIRV-Reflect#249]
It seems like spirv-reflect is correctly assigning the SPV_REFLECT_DECORATION_WEIGHT_TEXTURE, but it never assigns the correct descriptor type in these instances, SPV_REFLECT_DESCRIPTOR_TYPE_SAMPLE_WEIGHT_IMAGE_QCOM.

**Comment by spencer-lunarg** [comment]
@raegnar do you have an example SPIR-V file to help look at the problem

**Comment by randallrvr** [comment]
Here is fairly minimal spir-v file (renamed as a txt so it could be uploaded)
[sample_qcom_spv.txt](https://github.com/KhronosGroup/SPIRV-Reflect/files/14046365/sample_qcom_spv.txt)

**Comment by randallrvr** [comment]
Just following up, was that what you needed?

**Comment by spencer-lunarg** [comment]
yes the spirv is good, I just haven't had time recently to work on SPIRV-Reflect
